fix(autofix): smooth output stream

roaga opened the pull request:
- instead of abruptly resetting when a new stream starts, we now just append new text so it's less jarring
- changes the shade of pink for the progress bar to be more visible in dark mode and chonk
- adjusts the insight card animation to look less weird, and also include the typing animation
- speed up all the typing animations

roaga committed:
UI fixes

jennmueng commented on static/app/components/events/autofix/autofixInsightCards.tsx:
```diff
@@ -117,6 +119,31 @@ function AutofixInsightCard({
   const insightCardAboveIndex = index - 1 >= 0 ? index - 1 : null;
 
+  // Typing animation for new insight
+  useEffect(() => {
+    if (!isNewInsight) {
+      setDisplayedInsightTitle(insight.insight);
+      titleIndexRef.current = insight.insight.length;
+      return () => {};
+    }
+
+    setDisplayedInsightTitle('');
+    titleIndexRef.current = 0;
+
+    const intervalId = window.setInterval(() => {
+      if (titleIndexRef.current < insight.insight.length) {
+        setDisplayedInsightTitle(insight.insight.slice(0, titleIndexRef.current + 1));
+        titleIndexRef.current++;
+      } else {
+        window.clearInterval(intervalId);
+      }
+    }, 15);
+
+    return () => {
+      window.clearInterval(intervalId);
+    };
+  }, [insight.insight, isNewInsight]);
```
thinking out loud here, but we use a typing animation like this in multiple places, might be worth making a component with this logic that can be reusable

roaga committed:
Reusable hook

jennmueng commented on static/app/components/events/autofix/autofixOutputStream.tsx:
```diff
@@ -82,50 +87,57 @@ export function AutofixOutputStream({
+      }
     };
-  }, [displayedActiveLog, activeLog]);
+  }, [stream, displayedText]);
```
seems fragile to depend on displayedText when we're setting displayedText in here as well...and I wonder if we can use the new typing animation hook here too?

roaga replied:
I don't think the hook can be re-used here because this is trying to manage and accumulate multiple constantly changing stream texts vs. just animating one static piece of text.

And the dependency on displayedText is needed for the reset logic when combining multiple streams

roaga committed:
feat(autofix): Add support for multiline input (#89973)

Switch all user inputs in Autofix to be TextArea's so users can use
shift + enter to add multiline text. No visible changes by default
though.

Also add some truncation logic to insight cards to show just the first
line of multiline cards by default, and show the rest when expanded.
